happypose_ros/test/test_single_view_icp.py

```diff
@@ -0,0 +1,190 @@
+#!/usr/bin/env python
+
+import numpy as np
+import PIL.Image
+import time
+import pytest
+import torch
+
+import launch_ros.actions
+import launch_testing.actions
+import launch_testing.markers
+
+from launch import LaunchDescription
+from launch.substitutions import PathJoinSubstitution
+from launch_ros.substitutions import FindPackageShare
+
+from happypose_testing_utils import create_camera_reliable_qos_config
+
+
+from geometry_msgs.msg import Point, Pose, Quaternion
+from sensor_msgs.msg import Image
+
+from ament_index_python.packages import get_package_share_directory
+
+from launch_testing.io_handler import ActiveIoHandler
+
+from happypose_testing_utils import (
+    HappyPoseTestCase,
+    assert_bbox,
+    assert_and_find_detection,
+    assert_pose_equal,
+)
+
+
+class TestHappyposeSingleViewNode(HappyPoseTestCase):
+    @classmethod
+    def setUpClass(cls) -> None:
+        """
+
+        :raises unittest.SkipTest: Used to prevent the base class from executing as a separate test case.
+        """
+
+        super().setUpClass(
+            [("cam_1", Image, True)],
+            "test_single_view_icp",
+        )
+        image_path = get_package_share_directory("happypose_ros") + "/test"
+        cls.rgb = np.asarray(PIL.Image.open(image_path + "/rgb/000629.png"))
+        # https://github.com/thodan/bop_toolkit/blob/master/docs/bop_datasets_format.md
+        # "Multiply the depth image with this factor to get depth in mm"
+        # ycbv "depth_scale" is 0.1
+        scale = 0.1 / 1000  # image units -> meters
+        cls.depth = scale * np.array(
```

Comment:
```suggestion
        cls.depth = 1e-3 * np.array(
```

Reply:
1e-4 but I would keep the cumbersome 0.1 / 1000 for clarity